ocaml/xapi/repository_helpers.ml

```diff
@@ -173,12 +179,14 @@ module Guidance = struct
         let msg = error_msg l in
         raise Api_errors.(Server_error (internal_error, [msg]))
 
-  let resort_guidances gs =
-    match GuidanceStrSet.find_opt "RebootHost" gs with
-    | Some _ ->
-        GuidanceStrSet.singleton "RebootHost"
-    | None ->
+  let resort_guidances ~kind gs =
+    match (find_opt RebootHost gs, kind) with
+    | Some _, _ ->
+        singleton RebootHost
+    | None, Recommended ->
         gs
+    | None, Absolute ->
+        filter (fun g -> g != EvacuateHost) gs
```

Comment:
You should use `<>` (`!=` would have probably worked this time, but only by chance as the type is a very simple variant type).